### PR TITLE
Up proc-macro2 to 1.20

### PR DESCRIPTION
**Pull request by kjeremy**
This changes the way Display is implemented

### CODE REVIEW

**Comment by kjeremy** [comment]
I think this is right (it compiles, passes tests and runs) but I'm unfamiliar with this section of the codebase.

File: xtask/src/codegen/gen_syntax.rs [hunk 1]
```diff
@@ -247,14 +247,14 @@ fn generate_nodes(kinds: KindsSrc<'_>, grammar: &AstSrc) -> Result<String> {
         #(#display_impls)*
     };
 
-    let ast = ast.to_string().replace("T ! [ ", "T![").replace(" ] )", "])");
```
Comment (matklad) [inline xtask/src/codegen/gen_syntax.rs:250]: CAn we keep `T ! [` -> `T![` transform?

Comment (kjeremy) [reply xtask/src/codegen/gen_syntax.rs:250]: Sure thing. It just didn't seem to do anything.

Comment (kjeremy) [reply xtask/src/codegen/gen_syntax.rs:250]: oh I see why...

**Comment by matklad** [comment]
Yeah, much better now :-)

bors r+

Thanks!

**Pull request merged**